gtclang/src/gtclang/Support/ClangCompat/FileUtil.h

```diff
@@ -0,0 +1,41 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#ifndef GTCLANG_SUPPORT_CLANGCOMPAT_FILE_UTIL_H
+#define GTCLANG_SUPPORT_CLANGCOMPAT_FILE_UTIL_H
+
+#include "clang/Basic/FileManager.h"
+#include "clang/Basic/Version.h"
+
+namespace gtclang {
```

Comment:
```suggestion
namespace gtclang::clang_compat::FileUtil {
```
I propose that for new files, we slowly use C++17 features, and at some point we can do a cleanup round for old features.